### PR TITLE
Add DeleteAttachment

### PR DESCRIPTION
**Pull request by edwinakatosh**

### CODE REVIEW

File: issue_test.go [hunk 1]
```diff
@@ -538,6 +538,26 @@ func TestIssueService_PostAttachment_NoAttachment(t *testing.T) {
...
+
+	resp, err := testClient.Issue.DeleteAttachment("10054")
+	if resp.StatusCode != 204 {
+		t.Error("Expected attachment not deleted.")
+	}
+	if err != nil {
+		t.Errorf("Error given: %s", err)
+	}
+}
+
```
Comment (ghostsquad) [inline issue_test.go]: Can you please add negative tests? to ensure that the `if` branches for error checking are validated?

Comment (edwinakatosh) [reply issue_test.go]: So when err == nil or StatusCode == 404, 403?

Please add edits so I can better understand.

**Comment by ghostsquad** [comment]
Yes please, that exactly.

Cheers,
Wes

**Comment by edwinakatosh** [comment]
Updated.

> Yes please, that exactly. Cheers, Wes
> […](#)

**Comment by ghostsquad** [comment]
@edwinakatosh I apologize for the delay in getting to this PR. Can you rebase & squash your commits in this PR while following this standard for your commit message?

https://www.conventionalcommits.org/en/v1.0.0-beta.3/

This will allow me to rebase & merge, and provide nice changelog and releases.

**Comment by edwinakatosh** [comment]
@ghostsquad please verify

**Pull request merged**